### PR TITLE
8339810: Clean up the code in sun.tools.jar.Main to properly close resources and use ZipFile during extract

### PR DESCRIPTION
**Pull request by jaikiran**
Can I please get a review of this change which proposes to address https://bugs.openjdk.org/browse/JDK-8339810?

As noted in the issue we have a few places in the jar's tool `Main` class where we currently don't close the resources in a try/finally block. The commit in this PR updates the relevant places to use a try-with-resources. Additionally, in the extract() implementation of the `Main` class, we use the `ZipFile` when a JAR file is being extracted. This matches with what we do in the rest of the code in that `Main` class where a jar tool operation is a being run against a file.

No new test has been added given the nature of this change and existing tests in `test/jdk/tools/jar` continue to pass with this change. tier1, tier2 and tier3 testing is currently in progress.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8339810](https://bugs.openjdk.org/browse/JDK-8339810): Clean up the code in sun.tools.jar.Main to properly close resources and use ZipFile during extract (**Bug** - P4)

### Reviewers
 * [Lance Andersen](https://openjdk.org/census#lancea) (@LanceAndersen - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/20928/head:pull/20928` \
`$ git checkout pull/20928`

Update a local copy of the PR: \
`$ git checkout pull/20928` \
`$ git pull https://git.openjdk.org/jdk.git pull/20928/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 20928`

View PR using the GUI difftool: \
`$ git pr show -t 20928`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/20928.diff">https://git.openjdk.org/jdk/pull/20928.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/20928#issuecomment-2339661922)

### CODE REVIEW

File: src/jdk.jartool/share/classes/sun/tools/jar/Main.java [hunk 1]
```diff
@@ -1524,12 +1506,12 @@ void list(InputStream in, String files[]) throws IOException {
      * Lists contents of JAR file, via ZipFile.
      */
     void list(String fname, String files[]) throws IOException {
```
Comment (sormuras) [inline src/jdk.jartool/share/classes/sun/tools/jar/Main.java:1508]:
```suggestion
    void list(String fname, String[] files) throws IOException {
```

Synchronize with the array syntax change in (new) line 1490.

Comment (jaikiran) [reply src/jdk.jartool/share/classes/sun/tools/jar/Main.java]: You are right - I missed this line and a few other similar declarations. I've now updated the PR to address them.

**Review by LanceAndersen — APPROVED** [review]
Hi Jai,

Thank you for the cleanup, I think the changes look good

**Pull request closed without merging**